Procfile

```diff
@@ -0,0 +1 @@
+web: node index.js
```

Comment:
I don't think you need a procfile for the deployment of a react app on Heroku, what you need to do is to set up serve and add a Heroku post-build script.
Look into that